OPHKIOS-244: hyväksy/hylkää modaalit

saku-koodari opened the pull request:
## Yhteenveto
https://jira.eduuni.fi/browse/OPHKIOS-244 - tehtävän

Uudet modaalit. Ei ole täysin figma-speksien mukainen, mutta voin viilata tyyliä vielä jos halutaan
<img width="666" height="423" alt="image" src="https://github.com/user-attachments/assets/c70bb5fc-8cc4-40f8-b7aa-74f0371c1217" />

## Lisätiedot

Ei kutsu bäkkäriä, vaan MSW:llä mokataan `PUT api/v1/clerk/registration/approval`

## Huomautukset

_Muut mahdolliset huomautukset..._

## Check-lista

- [x] yarn.lock päivitetty
- [ ] Käännösten Excel-tiedosto päivitetty
- [ ] UI-muutoksia testattu eri selaimilla ja mobiilinäkymässä

saku-koodari committed:
YKI(Frontend): Add free registration modal

kimmotaskinen-knowit commented on frontend/packages/shared/src/components/CustomModal/CustomModal.tsx:
```diff
@@ -1,13 +1,13 @@
 
 type CustomModalProps = ModalProps & {
-  modalTitle?: string;
+  modalTitle?: ReactNode | string;
```
Tämä on ihan hyvä idea kyllä mutta kysyin vielä Artolta halutaanko toi sulkemisruksi tonne myös ton “Peruuta” napin lisäksi. Mutta jos pidetään toi ruksi niin tarvii tehdä pari pientä steppiä kun toi shared käyttää versioituja paketteja. Mut tsekataan ne erikseen jos tulee tarvis

kimmotaskinen-knowit replied:
Todettiin et otetaan toi ruksi mukaan

kimmotaskinen-knowit commented on frontend/packages/yki/public/i18n/en-GB/public.json:
```diff
@@ -125,7 +125,15 @@
+            "subTitle": "Olet hyväksymässä ilmoittautumisen maksuttomuuden."
           }
+        },
+        "toasts": {
```
Nämä tarvinee laittaa tuonne ylemmän “toasts” alle jos katsoo tuota translation pathia itse komponentissa

kimmotaskinen-knowit replied:
ja lisää tuonne fi ja sv puolella kans

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkFreeRegistration/FreeRegistrationDetails.tsx:
```diff
@@ -73,6 +77,24 @@ export const ClerkFreeRegistrationDetails = () => {
+    }
+  }, [registrationDetails, approvalStatus, showToast, t]);
+
+  const [modalOpen, handleModal] = React.useState(false);
```
booleaneissa pyritty pitään se tieto noissa nimissä, esim: isModalOpen, setIsModalOpen

Ja toi useState näyttäs olevan jo importattuna ni ei tarvi React.

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkFreeRegistration/FreeRegistrationDetails.tsx:
```diff
@@ -90,9 +112,60 @@ export const ClerkFreeRegistrationDetails = () => {
+            }
+          >
+            <div className="rows gapped">
+              <Text className="margin-top" style={{ fontSize: '0.8em' }}>
```
Design näyttäs default 16px olevan toi fonttikoko ni tän voinee poistaa.

em ollaan aika harvakseltaan käytetty kun se compoundaa niin tuntunut seiffimmältä ajaa vaan rem

saku-koodari replied:
Laitoin sen, koska ilman `0.8em` se näyttää tältä
<img width="580" height="227" alt="image" src="https://github.com/user-attachments/assets/c5b01ffd-0dc5-46b8-8996-aaea0c9a55b3" />

ja visuaalisesti figmaspeksi näytti tältä
<img width="798" height="257" alt="image" src="https://github.com/user-attachments/assets/0dfad521-7be4-497f-ba60-ff32c73d8974" />


___

Ehkä nyt kun mietin ja katson tuota speksiä, niin tuo button-ryhmä vois sisältää 20px enemmän ja koko modaali 300 px leveämpi. Ja en tehnyt sitä leveyyden lisäyttä alun perin, koska siihen ei tainnut olla olemassa olevia css-luokkia, vaan olisi pitäny tehdä sekin itse.

Edit: Poistin noi `0.8em` - säännöt

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkFreeRegistration/FreeRegistrationDetails.tsx:
```diff
@@ -90,9 +112,60 @@ export const ClerkFreeRegistrationDetails = () => {
+                  data-testid="freeregistration-modal__cancel"
+                  variant={Variant.Outlined}
+                  color={Color.Primary}
+                  style={{ fontSize: '0.8em' }}
```
sama ku yllä eli ei tarvine

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkFreeRegistration/FreeRegistrationDetails.tsx:
```diff
@@ -90,9 +112,60 @@ export const ClerkFreeRegistrationDetails = () => {
+                  data-testid="freeregistration-modal__approve"
+                  variant={Variant.Contained}
+                  color={Color.Primary}
+                  style={{ fontSize: '0.8em' }}
```
sama ku yllä eli ei tarvine

kimmotaskinen-knowit commented on frontend/packages/yki/src/redux/sagas/clerkFreeRegistration.ts:
```diff
@@ -24,9 +27,20 @@ function* loadClerkFreeRegistrationsSaga() {
 
+function* approveFreeRegistrationSaga() {
+  try {
+    yield call(axiosInstance.put, '/api/v1/clerk/registration/approval');
```
tän voi pukkaa tonne APIEndpoints enumiin

kimmotaskinen-knowit commented on frontend/packages/yki/src/tests/msw/handlers.ts:
```diff
@@ -188,4 +188,7 @@ export const handlers = [
       return notFound();
     }
   }),
+  http.put('/api/v1/clerk/registration/approval', () => {
```
Jahas, oon ite näyttänyt tässä huonoa esimerkkiä, nappaa tääki tuolta APIEnpoints :)

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkFreeRegistration/FreeRegistrationDetails.tsx:
```diff
@@ -90,9 +112,60 @@ export const ClerkFreeRegistrationDetails = () => {
           >
             {t('details.buttons.approveFreeRegistration')}
           </OphButton>
+          <CustomModal
```
Tän vois heittään tonne render runkoon alla ja pitää tää nappien render vaan puhtaasti nappien renderiä varten.

Nyt kun tohon tulee vielä toinen modaali niin ehkä jopa paras nappaa nää modaalit omaksi komponenteiksi niin pysyy vähän siistimpänä tää

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkFreeRegistration/FreeRegistrationDetails.tsx:
```diff
@@ -90,9 +112,60 @@ export const ClerkFreeRegistrationDetails = () => {
+                  {translateCommon('cancel')}
+                </CustomButton>
+                <CustomButton
+                  data-testid="freeregistration-modal__approve"
```
näitä ei näissä napeissa taideta käyttää mihinkään ku cypress targetoidaan role ja name combolla?

saku-koodari replied:
joo, ei käytetä

saku-koodari committed:
YKI(Frontend) PR fixes:
 - fix naming convention
 - put toast i18n value to the correct place
 - remove custom fontSize resizes
 - put the api endpoint to enums
 - refactor modal: save it to a new file
   - and relocate the modal into render root

saku-koodari committed:
YKI(Frontend): Add reject modal

saku-koodari committed:
YKI(Frontend): [deploy] Bump shared package to version 1.14 and update snapshots

saku-koodari committed:
YKI(Frontend): PR fixes part 2:
 - Use H1 for the modal title
 - remove data-testid
 - remove linting warnings
 - make close icon large by using large font size
 - fix modal naming
 - wrap text around h1 tag

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkFreeRegistration/FreeRegistrationDetails.tsx:
```diff
@@ -47,6 +52,9 @@ export const ClerkFreeRegistrationDetails = () => {
   const translateLevel = (level: string) =>
     translateCommon('languageLevel.' + level);
 
+  const [isApproveModalOpen, setIsApproveModal] = useState(false);
```
setIsApproveModalOpen ja sama toiseen

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkFreeRegistration/FreeRegistrationModal.tsx:
```diff
@@ -0,0 +1,132 @@
+            alignItems="flex-start"
+            gap={1}
+          >
+            {t('details.modals.reject.title')}
```
Tän voi kans wrappaa H1 tagiin

kimmotaskinen-knowit reviewed (approved):
Hyvää työtä!

saku-koodari committed:
YKI(Frontned): Fix test
